### PR TITLE
Adding version pinning and dependency download.

### PR DESCRIPTION
**Pull request by shotah**
AC: Able to supply information to download a specific version of a mod.
AC: Ability to download any required dependencies listed by the mod.

Concessions:
* Version pinning
**  Forge does not track a mod version, using file name instead.
* Dependency Download
** Forge does not track FileId like it is supposed to, fileId references current mod, not dependency mod. (Code is left in if bug is fixed in forge.)
** Defaulted to just looking up mod and releaseType of Release, this could cause an error if dependency does not support MC version or is not in a Release state.

### CODE REVIEW

**Review by itzg — COMMENTED** [review]
Just a wording tweak/question in the README

File: README.md [hunk 1]
```diff
@@ -768,7 +769,13 @@ Example of expected forge api project ids, releases, and key:
   MODS_FORGEAPI_KEY: $WRX...
 ```
 
-Example of expected ForgeAPI file format: **Note**: name is currently unused, but can be used to document each entry.
+Example of expected ForgeAPI file format.
+
+**Field Description**: 
+* Name is currently unused, but can be used to document each entry.
+* Project id is the id found on the CurseForge website for a particular mod
+* Release Type corresponds to R, B, A on each of the files. Default is R for Release.
```
[the remark this answers is not in the document]
Comment (shotah) [reply README.md:777]: I wish it was! Sadly that isn't how forge denotes it on their website.

image

[the remark this answers is not in the document]
Comment (shotah) [reply README.md:777]: Phrasing needs to edited I'd agree, rereading I'm not calling out that you need to type Release|Beta|Alpha.

**Review by itzg — APPROVED** [review]
Thanks again!

**Pull request merged**